connection.go

```diff
@@ -264,6 +293,33 @@ func (sc *snowflakeConn) QueryContext(ctx context.Context, query string, args []
 			RowSetBase64: data.Data.RowSetBase64,
 		},
 	}
+
+	if sc.isMultiStmt(data.Data) {
+		childResults, err := getChildResults(data.Data.ResultIDs, data.Data.ResultTypes)
+		if err != nil {
+			return nil, err
+		}
+		var nextChunkDownloader *snowflakeChunkDownloader
+		firstResultSet := false
+
+		for _, child := range childResults {
+			resultPath := fmt.Sprintf("/queries/%s/result", child.id)
+			childData, err := sc.getQueryResult(ctx, resultPath)
+			if err != nil {
+				return nil, err
+			}
+			if !firstResultSet {
+				// populate rows.ChunkDownloader with the first child
+				rows.ChunkDownloader = populateChunkDownloader(ctx, sc, childData.Data)
+				nextChunkDownloader = rows.ChunkDownloader
+				firstResultSet = true
+			} else {
+				nextChunkDownloader.NextDownloader = populateChunkDownloader(ctx, sc, childData.Data)
+				nextChunkDownloader = nextChunkDownloader.NextDownloader
+			}
+		}
+	}
+
```

Comment:
nit; can this be in a separate function?

Reply:
Yes, though the function call to it will be a little ugly. Would you recommend I do the same for `ExecContext()`as well to mirror this?

Reply:
A question is code readability. In general, deeply indented code is often hard to read. Not critical. If adding a function is ugly, I'm ok too.